README
NAME
    Attribute::SubName - Naming anonymous subroutines via attributes

VERSION
    version 1.100710

SYNOPSIS
        use Attribute::SubName;
        my $coderef = sub :Name(foo) { print "got: $_\n"; };
        print foo("hi");

DESCRIPTION
    This module provides an attribute ":Name" that you can use on anonymous
    subroutines to give them a name. This is useful as they will then show
    up with that name in stack traces (cf. Carp). The naming is done with
    Sub::Name. Additionally, the attribute also installs the newly named
    subroutine in the proper glob slot so you can refer to it by name.

INSTALLATION
    See perlmodinstall for information and options on installing Perl
    modules.

BUGS AND LIMITATIONS
    No bugs have been reported.

    Please report any bugs or feature requests through the web interface at
    <http://rt.cpan.org/Public/Dist/Display.html?Name=Attribute-SubName>.

AVAILABILITY
    The latest version of this module is available from the Comprehensive
    Perl Archive Network (CPAN). Visit <http://www.perl.com/CPAN/> to find a
    CPAN site near you, or see
    <http://search.cpan.org/dist/Attribute-SubName/>.

    The development version lives at
    <http://github.com/hanekomu/Attribute-SubName/>. Instead of sending
    patches, please fork this project using the standard git and github
    infrastructure.

AUTHOR
      Marcel Gruenauer <marcel@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2007 by Marcel Gruenauer.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Perl extension Attribute-SubName

1.100710  2010-03-12 15:15:29 Europe/Vienna
    - converted the distribution to Dist::Zilla-style

0.04  Thu Jul 25 15:27:43 CEST 2008 (Marcel Gruenauer <marcel@cpan.org>)
     - Changed dist style and Changes back to standard. People didn't like it -
       the nail that sticks out gets hammered down.
     - Added standard test files; this will also help with CPANTS.

0.03  Thu, 29 May 2008 12:00:21 -0000 (Marcel Gruenauer <marcel@cpan.org>)
     - set the version to 0.03
     - accept attribute data as scalar and array ref now to please both
       Attribute::Handlers 0.78 and 0.79
     - .shipit: fixed svk.tagpattern

0.02  Mon, 03 Dec 2007 23:15:40 -0000 (Marcel Gruenauer <marcel@cpan.org>)
     - set the version to 0.02
     - Makefile.PL: requires Sub::Name

0.01  Mon, 03 Dec 2007 21:59:25 -0000 (Marcel Gruenauer <marcel@cpan.org>)
     - original version


lib/Attribute/SubName.pm
use 5.006;
use strict;
use warnings;

package Attribute::SubName;
our $VERSION = '1.100710';
# ABSTRACT: Naming anonymous subroutines via attributes

use Sub::Name;
use base 'Attribute::Handlers';

sub UNIVERSAL::Name : ATTR(CODE) {
    my ($package, $symbol, $referent, $attr, $data, $phase) = @_;
    $data = [$data] unless ref $data eq 'ARRAY';
    for my $item (@$data) {
        my $name = "${package}::${item}";
        subname $name => $referent;
        no strict 'refs';
        *{$name} = $referent;
    }
}
1;


__END__
=pod

=head1 NAME

Attribute::SubName - Naming anonymous subroutines via attributes

=head1 VERSION

version 1.100710

=head1 SYNOPSIS

    use Attribute::SubName;
    my $coderef = sub :Name(foo) { print "got: $_\n"; };
    print foo("hi");

=head1 DESCRIPTION

This module provides an attribute C<:Name> that you can use on anonymous
subroutines to give them a name. This is useful as they will then show up with
that name in stack traces (cf. L<Carp>). The naming is done with L<Sub::Name>.
Additionally, the attribute also installs the newly named subroutine in the
proper glob slot so you can refer to it by name.

=head1 INSTALLATION

See perlmodinstall for information and options on installing Perl modules.

=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests through the web interface at
L<http://rt.cpan.org/Public/Dist/Display.html?Name=Attribute-SubName>.

=head1 AVAILABILITY

The latest version of this module is available from the Comprehensive Perl
Archive Network (CPAN). Visit L<http://www.perl.com/CPAN/> to find a CPAN
site near you, or see
L<http://search.cpan.org/dist/Attribute-SubName/>.

The development version lives at
L<http://github.com/hanekomu/Attribute-SubName/>.
Instead of sending patches, please fork this project using the standard git
and github infrastructure.

=head1 AUTHOR

  Marcel Gruenauer <marcel@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2007 by Marcel Gruenauer.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



Makefile.PL

use strict;
use warnings;

 BEGIN { require 5.006; } 

use ExtUtils::MakeMaker 6.11;



my %WriteMakefileArgs = (
                       'test' => {
                                   'TESTS' => 't/*.t'
                                 },
                       'NAME' => 'Attribute::SubName',
                       'DISTNAME' => 'Attribute-SubName',
                       'CONFIGURE_REQUIRES' => {
                                                 'ExtUtils::MakeMaker' => '6.11'
                                               },
                       'AUTHOR' => 'Marcel Gruenauer <marcel@cpan.org>',
                       'BUILD_REQUIRES' => {},
                       'ABSTRACT' => 'Naming anonymous subroutines via attributes',
                       'EXE_FILES' => [],
                       'VERSION' => '1.100710',
                       'PREREQ_PM' => {
                                        'Test::More' => '0',
                                        'Scalar::Util' => '0',
                                        'Attribute::Handlers' => '0',
                                        'English' => '0',
                                        'File::Find' => '0',
                                        'File::Temp' => '0',
                                        'Sub::Name' => '0',
                                        'Carp' => '0'
                                      },
                       'LICENSE' => 'perl'
                     );


delete $WriteMakefileArgs{LICENSE}
  unless eval { ExtUtils::MakeMaker->VERSION(6.31) };

WriteMakefile(%WriteMakefileArgs);



